types/director/index.d.ts

```diff
@@ -0,0 +1,549 @@
+// Type definitions for director 1.2
+// Project: https://github.com/flatiron/director
+// Definitions by: pastelmind <https://github.com/pastelmind>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.4
+
+/// <reference types="node" />
+
+import { IncomingMessage, ServerResponse } from "http";
+
+/** Utility type */
+export type BaseOrArray<T> = T | readonly T[];
+
+/**
+ * Route handler callback.
+ *
+ * In synchronous mode, the handler is called with all matched tokens as
+ * arguments. If a handler returns `false`, the router will skip all remaining
```

Comment:
It was an arbitrary decision. I didn't want to constain the return type of handler callbacks, since the Router only cares about whether the returned value `=== false`. Any other value is simply ignored. I don't want to invalidate existing code that relies on this behavior.

Perhaps returning `any` or `unknown` might have been better? Ideas are welcome.